lib/bambora/headers.rb

```diff
@@ -1,17 +1,30 @@
 # frozen_string_literal: true
 
 module Bambora
+  ##
+  # Builds Headers for HTTP requests.
   class Headers
-    class << self
-      def build(api_key:, merchant_id:, sub_merchant_id: nil)
-        headers = { 'Authorization' => "Passcode #{passcode(merchant_id, api_key)}" }
-        headers['Sub-Merchant-Id'] = sub_merchant_id unless sub_merchant_id.nil?
-        headers
-      end
+    attr_reader :api_key, :merchant_id, :sub_merchant_id, :content_type
 
-      def passcode(merchant_id, api_key)
-        Base64.encode64("#{merchant_id}:#{api_key}").delete("\n")
+    def initialize(options = {})
+      options.each do |key, value|
+        instance_variable_set("@#{key}", value)
       end
     end
+
+    def build
```

Comment:
Yes, it is. Although, I normally see it as a class method. I tried both and the instance method gave me a tidier class. This might be unconventional for a builder though.